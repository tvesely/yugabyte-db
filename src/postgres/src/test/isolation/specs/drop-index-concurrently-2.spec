setup
{
	CREATE TABLE test_dc(id serial primary key, data int);
	CREATE INDEX test_dc_data ON test_dc(data);
}

session "s1"
setup { BEGIN; }
step "explain" { EXPLAIN (COSTS OFF) SELECT * FROM test_dc WHERE data=34343; }
step "rollback" { ROLLBACK; }
step "droptab" { DROP TABLE test_dc; }
step "selecti" { SELECT indexrelid::regclass, indisvalid, indisready FROM pg_index WHERE indexrelid = 'test_dc_data'::regclass; }
step "dropi" { DROP INDEX test_dc_data; }

session "s2"
step "drop" { DROP INDEX CONCURRENTLY test_dc_data; }

session "s3"
step "cancel" { SELECT pg_cancel_backend(pid) FROM pg_stat_activity WHERE query = 'DROP INDEX CONCURRENTLY test_dc_data;'; }

permutation "explain" "drop" "cancel" "rollback" "droptab" "selecti" "dropi"
