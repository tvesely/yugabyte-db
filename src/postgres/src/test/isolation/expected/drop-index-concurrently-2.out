Parsed test spec with 3 sessions

starting permutation: explain drop cancel rollback droptab selecti dropi
step explain: EXPLAIN (COSTS OFF) SELECT * FROM test_dc WHERE data=34343;
QUERY PLAN     

Bitmap Heap Scan on test_dc
  Recheck Cond: (data = 34343)
  ->  Bitmap Index Scan on test_dc_data
        Index Cond: (data = 34343)
step drop: DROP INDEX CONCURRENTLY test_dc_data; <waiting ...>
step cancel: SELECT pg_cancel_backend(pid) FROM pg_stat_activity WHERE query = 'DROP INDEX CONCURRENTLY test_dc_data;';
pg_cancel_backend

t              
step drop: <... completed>
error in steps cancel drop: ERROR:  canceling statement due to user request
step rollback: ROLLBACK;
step droptab: DROP TABLE test_dc;
step selecti: SELECT indexrelid::regclass, indisvalid, indisready FROM pg_index WHERE indexrelid = 'test_dc_data'::regclass;
indexrelid     indisvalid     indisready     

test_dc_data   f              f              
step dropi: DROP INDEX test_dc_data;
